File: Adapter/Dart/CurrentMediaPlayer.dart
import 'IMediaPlayer.dart';

class CurrentMediaPlayer implements IMediaPlayer {
  void playMKV(String filename) {
    print('Playing file $filename with MKV format');
  }

  void playMP4(String filename) {
    print('Playing file $filename with MP4 format');
  }
}

File: Adapter/Javascript/CurrentMediaPlayer.js
import IMediaPlayer from "./IMediaPlayer.js";

class CurrentMediaPlayer extends IMediaPlayer {
  playMKV(filename) {
    console.log(`Playing file ${filename} with MKV format`);
  }

  playMP4(filename) {
    console.log(`Playing file ${filename} with MP4 format`);
  }
}

export default CurrentMediaPlayer;